presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchanger.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.exchange;
+
+import com.facebook.presto.spi.Page;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import static com.facebook.presto.operator.Operator.NOT_BLOCKED;
+
+public interface LocalExchanger
+{
+    LocalExchanger FINISHED = new LocalExchanger()
+    {
+        @Override
+        public void accept(Page page) {}
+
+        @Override
+        public ListenableFuture<?> getNotFullFuture()
+        {
+            return NOT_BLOCKED;
+        }
+    };
+
+    void accept(Page page);
+
+    default void finishedWriting() {}
```

Comment:
Please add this when there is actual use for it (I guess in distributed sort PR).

Reply:
I wanted somebody from FB to validate it if the direction is OK.